src/crystal/system/unix/evented/arena.cr

```diff
@@ -229,29 +183,65 @@ class Crystal::Evented::Arena(T)
     end
   end
 
-  private def at(index : Int32) : Pointer(Entry(T))
-    (@buffer + index).to_unsafe
+  private def at(index : Int32, grow : Bool) : Pointer(Entry(T))
+    raise IndexError.new unless 0 <= index < @capacity
+
+    entries_per_block = BLOCK_BYTESIZE // sizeof(Entry(T))
```

Comment:
**suggestion:** `entries_per_block` is used in a lot of methods, maybe we could extract it into a private helper method? It should be calculated at compile time so there's not even a point in storing it in a local variable.

Reply:
I tried making it a constant, but we can't access a generic when defining constants:

```
In src/crystal/system/unix/evented/arena.cr:32:23

 32 | ENTRIES_PER_BLOCK = BLOCK_BYTESIZE // sizeof(Entry(T))
                          ^-------------
Error: undefined constant BLOCK_BYTESIZE
```

Let's make it a method instead :shrug: